docs/tutorial_ptk.rst

```diff
@@ -0,0 +1,139 @@
+.. _tutorial_ptk:
+
+***********************************************
+Tutorial: ``prompt_toolkit`` custom keybindings
+***********************************************
+
+Are you really jonesing for some special keybindings? We can help you out with
+that. The first time is free and so is every other time!
+
+.. warning:: This tutorial will let you hook directly into the
+             ``prompt_toolkit`` keybinding manager. It will not stop you from
+             rendering your prompt completely unusable, so tread lightly.
+
+
+Overview
+========
+
+The ``prompt_toolkit`` shell has a registry for handling custom keybindings. You
+may not like the default keybindings in xonsh, or you may want to add a new key
+binding.
+
+We'll walk you though how to do this using ``prompt_toolkit`` tools to define
+keybindings and warn you about potential pitfalls.
+
+All of the code below can be entered into your `xonshrc <xonshrc.html>`_
+
+Control characters
+==================
+
+We can't and won't stop you from doing what you want, but in the interest of a
+functioning shell, you probably shouldn't mess with the following keystrokes.
+Some of them are `ASCII control characters
+<https://en.wikipedia.org/wiki/Control_character#In_ASCII>`_ and _really_
+shouldn't be used. The others are used by xonsh and will result in some loss of
+functionality (in less you take the time to rebind them elsewhere).
+
+.. list-table::
+    :widths: 2 2 2
+    :header-rows: 1
+
+    * - Keystroke
+      - ASCII control representation
+      - Default commmand
+    * - ``Control J``
+      - ``<Enter>``
+      - Run command
+    * - ``Control I``
+      - ``<Tab>``
+      - Indent, autocomplete
+    * - ``Control R``
+      - 
+      - Backwards history search
+    * - ``Control Z``
+      - 
+      - SIGSTOP current job
+    * - ``Control C``
+      -
+      - SIGINT current job
+
+
+Useful imports
+==============
+
+There are a few useful ``prompt_toolkit`` tools that will help us create better
+bindings::
+
+    from prompt_toolkit.keys import Keys
```

Comment:
Yes, there is the double colon above, so this will be interreted as a code block, which defaults to Python